scalanet/src/io/iohk/scalanet/peergroup/InetPeerGroupUtils.scala

```diff
@@ -23,6 +26,12 @@ object InetPeerGroupUtils {
     }
   }
 
+  def toTask[T](f: => Cancelable)(implicit tag: ClassTag[T]): Task[Cancelable] = {
```

Comment:
is the `tag` used here?

Reply:
It doesn't compile without that since the generic function loses the type, there are 2 toTask function. I will check if any otherway to resolve this

Reply:
Adding unused type parameter and implicit argument are not a good way to solve the problem. You should probably rename this function to `cancellableToTask` or `taskFromCancellable`.

Reply:
What's more during more careful analysis I found that your implementation is wrong as it never fulfils the task. Let's take a look at example usage from Monix documentation:
```
def delayResult1[A](timespan: FiniteDuration)(thunk: => A): Task[A] =
  Task.create { (scheduler, cb) =>
    val c = scheduler.scheduleOnce(timespan)(cb(Try(thunk)))
    // We can simply return `c`, but doing this for didactic purposes!
    Cancelable(() => c.cancel())
  }
```

You can see that `cb` (callback) function is passed and when the computation is finished it has to be called with the result. Otherwise the Task is never considered finished.

Reply:
Plus third, final point: is that version of function even used anywhere? I have removed it and the project compiles, the tests pass.